src/certificate.js

```diff
@@ -94,10 +120,56 @@ function verifyCertificate(certificate) {
   return true;
 }
 
+Certificate.prototype.privacyFilter = function _privacyFilter(fields) {
+  this.certificate = _.cloneDeep(this.certificate);
+
+  const { evidence, privateEvidence } = this.certificate.badge;
+  const { type, saltLength } = this.certificate.badge.evidencePrivacyFilter;
+  if (!type) throw new Error("Privacy filter algorithm cannot be found");
+  if (!saltLength) throw new Error("Privacy salt length cannot be found");
+  if (type !== "SaltedProof") throw new Error("Unsupported privacy filter");
+
+  const valueToRemove = fields instanceof Array ? fields : [fields];
+
+  const flattenedEvidence = flatten(evidence);
+  const privateEvidences = {};
```

Comment:
Might want to rename this to avoid confusion with the original `privateEvidence`.